src/gas/determineGasFeeSuggestions.test.ts

```diff
@@ -0,0 +1,294 @@
+import determineGasFeeSuggestions from './determineGasFeeSuggestions';
+import {
+  unknownString,
+  GasFeeEstimates,
+  LegacyGasPriceEstimate,
+  EthGasPriceEstimate,
+  EstimatedGasFeeTimeBounds,
+} from './GasFeeController';
+
+/**
+ * Builds mock data for the `fetchGasEstimates` function. All of the data here is filled in to make
+ * the gas fee estimation code function in a way that represents a reasonably happy path; it does
+ * not necessarily match the real world.
+ *
+ * @returns The mock data.
+ */
+function buildMockDataForFetchGasEstimates(): GasFeeEstimates {
+  return {
+    low: {
+      minWaitTimeEstimate: 10000,
+      maxWaitTimeEstimate: 20000,
+      suggestedMaxPriorityFeePerGas: '1',
+      suggestedMaxFeePerGas: '10',
+    },
+    medium: {
+      minWaitTimeEstimate: 30000,
+      maxWaitTimeEstimate: 40000,
+      suggestedMaxPriorityFeePerGas: '1.5',
+      suggestedMaxFeePerGas: '20',
+    },
+    high: {
+      minWaitTimeEstimate: 50000,
+      maxWaitTimeEstimate: 60000,
+      suggestedMaxPriorityFeePerGas: '2',
+      suggestedMaxFeePerGas: '30',
+    },
+    estimatedBaseFee: '100',
+  };
+}
+
+/**
+ * Builds mock data for the `fetchLegacyGasPriceEstimates` function. All of the data here is filled
+ * in to make the gas fee estimation code function in a way that represents a reasonably happy path;
+ * it does not necessarily match the real world.
+ *
+ * @returns The mock data.
+ */
+function buildMockDataForFetchLegacyGasPriceEstimates(): LegacyGasPriceEstimate {
+  return {
+    low: '10',
+    medium: '20',
+    high: '30',
+  };
+}
+
+/**
+ * Builds mock data for the `fetchEthGasPriceEstimate` function. All of the data here is filled in
+ * to make the gas fee estimation code function in a way that represents a reasonably happy path; it
+ * does not necessarily match the real world.
+ *
+ * @returns The mock data.
+ */
+function buildMockDataForFetchEthGasPriceEstimate(): EthGasPriceEstimate {
+  return {
+    gasPrice: '100',
+  };
+}
+
+/**
+ * Builds mock data for the `calculateTimeEstimate` function. All of the data here is filled in to
+ * make the gas fee estimation code function in a way that represents a reasonably happy path; it
+ * does not necessarily match the real world.
+ *
+ * @returns The mock data.
```

Comment:
The function description is doing the heavy lifting here, rather than the return description. Which seems fine to me - they're shown together. Sometimes it's easier to explain things in the function description. The type can be derived from the TypeScript return type as well, so that'll be shown as well with Intellisense.

Was there something else missing that you had in mind?

Reply:
Oh, I guess maybe you meant the meaning of each part of the mock data. Yeah I guess that's fair. Arguably that's explained already in the function being mocked though :thinking: And that is in the description.